src/modules/modules.controller.ts

```diff
@@ -16,12 +18,14 @@ import { TimeSlotsService } from './timeslots/time-slots.service';
 export class ModulesController {
   constructor(private modulesService: ModulesService, private timeSlotService: TimeSlotsService) {}
 
+  @UserTypes(UserEnum.ADMIN)
   @ApiCommonResponse({ type: ModuleDTO })
   @Post('/')
   async create(@Request() req: RequestWithUser, @Body() module: CreateModuleDTO): Promise<ModuleDTO> {
     return await this.modulesService.create(module);
   }
 
+  @UserTypes(UserEnum.TEACHER, UserEnum.TEACHING_STUDENT)
```

Comment:
I don't think a teacher or student should be able to edit the module - that is Admin only